lib/Bio/EnsEMBL/DataCheck/Test/DataCheck.pm

```diff
@@ -278,16 +278,33 @@ sub row_subtotals {
   $sql2 = $sql1 if ! defined $sql2;
   $min_proportion = 1 if ! defined $min_proportion;
 
-  if ($sql1 !~ /^\s*SELECT\s+[^,]+\s*,\s*COUNT[^,]+FROM.+GROUP\s+BY/ms) {
-    die "Invalid SQL statement for subtotals. Must select a single column first, then a count.\n($sql1)";
-  }
-  if ($sql2 !~ /^\s*SELECT\s+[^,]+\s*,\s*COUNT[^,]+FROM.+GROUP\s+BY/ms) {
-    die "Invalid SQL statement for subtotals. Must select a single column first, then a count.\n($sql2)";
-  }
-
   my ( undef, $rows1 ) = _query( $dbc1, $sql1 );
   my ( undef, $rows2 ) = _query( $dbc2, $sql2 );
 
+  my $len1 = @$rows1;
```

Comment:
The t/TestDataCheck.t test is failing (see travis report https://travis-ci.org/github/Ensembl/ensembl-datacheck/builds/723484605), because it's using an incorrect SQL statement in order to generate an error which can then be checked, to see if it is raised appropriately.
But that 'incorrect' statement is returning a scalar not an arrayref, so this line is generating an untrapped error.
So, need to check here for an arrayref, and maybe add another test case to trap queries with more than two columns.